coffeecard/CoffeeCard.WebApi/Controllers/v2/TicketsController.cs

```diff
@@ -50,5 +47,27 @@ public async Task<ActionResult<List<TicketResponse>>> Get([FromQuery] bool inclu
 
             return Ok(await _ticketService.GetTickets(user, includeUsed));
         }
+
+        /// <summary>
+        /// Uses the ticket on the specified menu item
+        /// </summary>
+        /// <param name="ticketId">The id of the ticket to be used</param>
+        /// <param name="menuItemId">The id of the menu item to use the ticket on</param>
+        /// <returns>The ticket that was used</returns>
+        /// <response code="200">Successful request</response>
+        /// <response code="401">Invalid credentials</response>
+        /// <response code="403">Menu item cannot be redeemed by this ticket</response>
+        /// <response code="404">No ticket on account or menu item not found</response>
+        [ProducesResponseType(typeof(UsedTicketResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(MessageResponseDto), StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(MessageResponseDto), StatusCodes.Status404NotFound)]
+        [HttpPost("use/{ticketId:int}/{menuItemId:int}")]
+        public async Task<ActionResult<UsedTicketResponse>> UseTicket(int ticketId, int menuItemId)
+        {
+            var user = await _claimsUtilities.ValidateAndReturnUserFromClaimAsync(User.Claims);
+
+            return Ok(await _ticketService.UseTicketAsync(user, ticketId, menuItemId));
```

Comment:
UseTicketAsync does not take a TicketId but a ProductId

Reply:
Addressed in e93b3e189f2a16dde9c1b6eb34a2b6ab52fa58f2